python311Packages.dvc-objects: 3.0.3 -> 3.0.6

fabaff opened the pull request:
Diff: https://github.com/iterative/dvc-objects/compare/refs/tags/3.0.3...3.0.6

Changelog: https://github.com/iterative/dvc-objects/releases/tag/3.0.6

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fabaff commented:
Result of `nixpkgs-review pr 279989` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>dvc (python311Packages.dvc)</li>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>dvc.dist (python311Packages.dvc.dist)</li>
    <li>python310Packages.dvc</li>
    <li>python310Packages.dvc-azure</li>
    <li>python310Packages.dvc-azure.dist</li>
    <li>python310Packages.dvc-data</li>
    <li>python310Packages.dvc-data.dist</li>
    <li>python310Packages.dvc-gs</li>
    <li>python310Packages.dvc-gs.dist</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-http.dist</li>
    <li>python310Packages.dvc-objects</li>
    <li>python310Packages.dvc-objects.dist</li>
    <li>python310Packages.dvc-s3</li>
    <li>python310Packages.dvc-s3.dist</li>
    <li>python310Packages.dvc-ssh</li>
    <li>python310Packages.dvc-ssh.dist</li>
    <li>python310Packages.dvc.dist</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.dvclive.dist</li>
    <li>python311Packages.dvc-azure</li>
    <li>python311Packages.dvc-azure.dist</li>
    <li>python311Packages.dvc-data</li>
    <li>python311Packages.dvc-data.dist</li>
    <li>python311Packages.dvc-gs</li>
    <li>python311Packages.dvc-gs.dist</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-http.dist</li>
    <li>python311Packages.dvc-objects</li>
    <li>python311Packages.dvc-objects.dist</li>
    <li>python311Packages.dvc-s3</li>
    <li>python311Packages.dvc-s3.dist</li>
    <li>python311Packages.dvc-ssh</li>
    <li>python311Packages.dvc-ssh.dist</li>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.dvclive.dist</li>
  </ul>
</details>

fabaff commented:
Result of `nixpkgs-review pr 279989` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>python310Packages.dvc-s3</li>
    <li>python310Packages.dvc-s3.dist</li>
    <li>python311Packages.dvc-s3</li>
    <li>python311Packages.dvc-s3.dist</li>
  </ul>
</details>
<details>
  <summary>32 packages built:</summary>
  <ul>
    <li>dvc (python311Packages.dvc)</li>
    <li>dvc.dist (python311Packages.dvc.dist)</li>
    <li>python310Packages.dvc</li>
    <li>python310Packages.dvc-azure</li>
    <li>python310Packages.dvc-azure.dist</li>
    <li>python310Packages.dvc-data</li>
    <li>python310Packages.dvc-data.dist</li>
    <li>python310Packages.dvc-gs</li>
    <li>python310Packages.dvc-gs.dist</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-http.dist</li>
    <li>python310Packages.dvc-objects</li>
    <li>python310Packages.dvc-objects.dist</li>
    <li>python310Packages.dvc-ssh</li>
    <li>python310Packages.dvc-ssh.dist</li>
    <li>python310Packages.dvc.dist</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.dvclive.dist</li>
    <li>python311Packages.dvc-azure</li>
    <li>python311Packages.dvc-azure.dist</li>
    <li>python311Packages.dvc-data</li>
    <li>python311Packages.dvc-data.dist</li>
    <li>python311Packages.dvc-gs</li>
    <li>python311Packages.dvc-gs.dist</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-http.dist</li>
    <li>python311Packages.dvc-objects</li>
    <li>python311Packages.dvc-objects.dist</li>
    <li>python311Packages.dvc-ssh</li>
    <li>python311Packages.dvc-ssh.dist</li>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.dvclive.dist</li>
  </ul>
</details>

fabaff commented:
Result of `nixpkgs-review pr 279989` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>dvc (python311Packages.dvc)</li>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>dvc.dist (python311Packages.dvc.dist)</li>
    <li>python310Packages.dvc</li>
    <li>python310Packages.dvc-azure</li>
    <li>python310Packages.dvc-azure.dist</li>
    <li>python310Packages.dvc-data</li>
    <li>python310Packages.dvc-data.dist</li>
    <li>python310Packages.dvc-gs</li>
    <li>python310Packages.dvc-gs.dist</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-http.dist</li>
    <li>python310Packages.dvc-objects</li>
    <li>python310Packages.dvc-objects.dist</li>
    <li>python310Packages.dvc-s3</li>
    <li>python310Packages.dvc-s3.dist</li>
    <li>python310Packages.dvc-ssh</li>
    <li>python310Packages.dvc-ssh.dist</li>
    <li>python310Packages.dvc.dist</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.dvclive.dist</li>
    <li>python311Packages.dvc-azure</li>
    <li>python311Packages.dvc-azure.dist</li>
    <li>python311Packages.dvc-data</li>
    <li>python311Packages.dvc-data.dist</li>
    <li>python311Packages.dvc-gs</li>
    <li>python311Packages.dvc-gs.dist</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-http.dist</li>
    <li>python311Packages.dvc-objects</li>
    <li>python311Packages.dvc-objects.dist</li>
    <li>python311Packages.dvc-s3</li>
    <li>python311Packages.dvc-s3.dist</li>
    <li>python311Packages.dvc-ssh</li>
    <li>python311Packages.dvc-ssh.dist</li>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.dvclive.dist</li>
  </ul>
</details>

NickCao reviewed (commented):
dependency on `shortuuid` is dropped, and `funcy` is conditional on python version.

fabaff committed:
python311Packages.dvc-objects: 3.0.3 -> 3.0.6

Diff: iterative/dvc-objects@refs/tags/3.0.3...3.0.6

Changelog: https://github.com/iterative/dvc-objects/releases/tag/3.0.6

fabaff commented:
> dependency on shortuuid is dropped, and funcy is conditional on python version.

There are still some left-overs which requires `shortuuid` for the tests.

fabaff commented:
Result of `nixpkgs-review pr 279989` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>dvc (python311Packages.dvc)</li>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>dvc.dist (python311Packages.dvc.dist)</li>
    <li>python310Packages.dvc</li>
    <li>python310Packages.dvc-azure</li>
    <li>python310Packages.dvc-azure.dist</li>
    <li>python310Packages.dvc-data</li>
    <li>python310Packages.dvc-data.dist</li>
    <li>python310Packages.dvc-gs</li>
    <li>python310Packages.dvc-gs.dist</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-http.dist</li>
    <li>python310Packages.dvc-objects</li>
    <li>python310Packages.dvc-objects.dist</li>
    <li>python310Packages.dvc-s3</li>
    <li>python310Packages.dvc-s3.dist</li>
    <li>python310Packages.dvc-ssh</li>
    <li>python310Packages.dvc-ssh.dist</li>
    <li>python310Packages.dvc.dist</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.dvclive.dist</li>
    <li>python311Packages.dvc-azure</li>
    <li>python311Packages.dvc-azure.dist</li>
    <li>python311Packages.dvc-data</li>
    <li>python311Packages.dvc-data.dist</li>
    <li>python311Packages.dvc-gs</li>
    <li>python311Packages.dvc-gs.dist</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-http.dist</li>
    <li>python311Packages.dvc-objects</li>
    <li>python311Packages.dvc-objects.dist</li>
    <li>python311Packages.dvc-s3</li>
    <li>python311Packages.dvc-s3.dist</li>
    <li>python311Packages.dvc-ssh</li>
    <li>python311Packages.dvc-ssh.dist</li>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.dvclive.dist</li>
  </ul>
</details>

fabaff commented:
Result of `nixpkgs-review pr 279989` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>dvc (python311Packages.dvc)</li>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>dvc.dist (python311Packages.dvc.dist)</li>
    <li>python310Packages.dvc</li>
    <li>python310Packages.dvc-azure</li>
    <li>python310Packages.dvc-azure.dist</li>
    <li>python310Packages.dvc-data</li>
    <li>python310Packages.dvc-data.dist</li>
    <li>python310Packages.dvc-gs</li>
    <li>python310Packages.dvc-gs.dist</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-http.dist</li>
    <li>python310Packages.dvc-objects</li>
    <li>python310Packages.dvc-objects.dist</li>
    <li>python310Packages.dvc-s3</li>
    <li>python310Packages.dvc-s3.dist</li>
    <li>python310Packages.dvc-ssh</li>
    <li>python310Packages.dvc-ssh.dist</li>
    <li>python310Packages.dvc.dist</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.dvclive.dist</li>
    <li>python311Packages.dvc-azure</li>
    <li>python311Packages.dvc-azure.dist</li>
    <li>python311Packages.dvc-data</li>
    <li>python311Packages.dvc-data.dist</li>
    <li>python311Packages.dvc-gs</li>
    <li>python311Packages.dvc-gs.dist</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-http.dist</li>
    <li>python311Packages.dvc-objects</li>
    <li>python311Packages.dvc-objects.dist</li>
    <li>python311Packages.dvc-s3</li>
    <li>python311Packages.dvc-s3.dist</li>
    <li>python311Packages.dvc-ssh</li>
    <li>python311Packages.dvc-ssh.dist</li>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.dvclive.dist</li>
  </ul>
</details>
